frontend/app/components/timed-battery/component.js

```diff
@@ -2,9 +2,25 @@ import Component from '@ember/component'
 import { computed } from '@ember/object'
 
 export default Component.extend({
-  height: computed('percentage', function() {
-    return 351 * this.percentage
+  didRender() {
+    this._super(...arguments)
+    this.animate
+  },
+
+  animate: computed('percentage', function() {
+    let element = document.getElementById(`battery-body-${this.index}`)
+    element.animate(
+      [{ height: '170px' }, { height: `${100 * this.percentage}%` }],
+      {
+        delay: 200,
+        duration: 2000,
+        easing: 'ease-in-out',
+        fill: 'forwards'
+      }
+    )
+    element.style.backgroundColor = this.color
   }),
+
```

Comment:
abusing a cp as observer is not really a good idea. even observers are not a good idea. you should only use them if you have other choice!! In this case a cp makes no sense since 1. the percentage never changes, 2. even if it changes we dont replay the animation. you can just set the animate in the init imo.

Reply:
And the color cp?